### PR TITLE
[MoM] Add nospell mutation to the anchoring crown

### PR DESCRIPTION
**Pull request by SariusSkelrets**
#### Summary
Mods "[MoM] Wearing the anchoring crown suppresses your spellcasting as intended"

#### Purpose of change

According to the comment in the anchoring crown's code, it's supposed to shut down spellcasting but doesn't since there's no way to do that

Turns out there's a way to do that

#### Describe the solution

Wearing the crown now gives a mutation disabling all spellcasting until you take it off

#### Describe alternatives you've considered

Tried to place the NO_SPELLCASTING flag directly on the crown but it didn't worked

#### Testing

Wear crown, try to cast spells, can't cast spells

Remove crown, try to cast spells, spells are cast

#### Additional context

None

### CODE REVIEW

File: data/mods/MindOverMatter/items/armor/head.json [hunk 1]
```diff
@@ -15,6 +14,7 @@
     "color": "blue",
     "material_thickness": 1,
     "flags": [ "DIMENSIONAL_ANCHOR" ],
+    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "mutations": [ "ANCHORCROWN_NOSPELL" ] } ] },
```
Comment (GuardianDll) [inline data/mods/MindOverMatter/items/armor/head.json:17]: Are you sure it works? I recall a longstanding issue, that mutations, given by enchantment (which relic_data is) is shown in @ menu, but have no effect whatsoever

**Comment by Standing-Storm** [comment]
Could you also document the NO_SPELLCASTING flag, please? It’s not in any of the docs it looks like

File: data/mods/MindOverMatter/items/armor/head.json (hunk 1 above)
Comment (SariusSkelrets) [reply data/mods/MindOverMatter/items/armor/head.json:17]: I did test it and it worked. Wearing the crown prevented me from using any spell at all, then removing it gave me my spellcasting back

Comment (Standing-Storm) [reply data/mods/MindOverMatter/items/armor/head.json:17]: Working on #68140, it looks like enchantment mutations given by items work, but enchantment mutations given by effects do not.

**Pull request merged**